cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -109,33 +109,41 @@ def test_verify_revert_snapshot(test_vm, provider, soft_assert, register_event,
         'password': credentials[provider.data['full_template']['creds']]['password'],
         'hostname': ip
     }
-    with SSHClient(**ssh_kwargs) as ssh_client:
-        ssh_client.run_command('touch snapshot1.txt')
-        snapshot1.create()
-        ssh_client.run_command('touch snapshot2.txt')
-        snapshot2 = new_snapshot(test_vm)
-        snapshot2.create()
-        snapshot1.revert_to()
+    ssh_client = SSHClient(**ssh_kwargs)
+    # We need to wait for ssh to become available on the vm, it can take a while. Without
+    # this wait, the ssh command would fail with 'port 22 not available' error.
+    # Easiest way to solve this is just mask the exception with 'handle_exception = True'
+    # and wait for successful completition of the ssh command.
+    # The 'fail_func' ensures we close the connection that failed with exception.
+    # Without this, the connection would hang there and wait_for would fail with timeout.
+    wait_for(lambda: ssh_client.run_command('touch snapshot1.txt').rc == 0, num_sec=300,
+             delay=20, handle_exception=True, fail_func=ssh_client.close())
+    snapshot1.create()
+    ssh_client.run_command('touch snapshot2.txt')
+    snapshot2 = new_snapshot(test_vm)
+    snapshot2.create()
+    snapshot1.revert_to()
     # Wait for the snapshot to become active
     logger.info('Waiting for vm %s to become active', snapshot1.name)
     wait_for(snapshot1.wait_for_snapshot_active, num_sec=300, delay=20, fail_func=sel.refresh)
     test_vm.wait_for_vm_state_change(desired_state=test_vm.STATE_OFF, timeout=720)
-    register_event(target_type='VmOrTemplate', target_name=test_vm.name,
```

Comment:
I agree that you don't need to check these events. But it would be good to check that snapshot related events come